ansible_ai_connect/ai/api/model_client/wca_client.py

```diff
@@ -451,7 +451,7 @@ def get_api_key(self, user, organization_id: Optional[int]) -> str:
             settings.ANSIBLE_AI_ENABLE_ONE_CLICK_TRIAL
             and any(up.is_active for up in user.userplan_set.all())
             and user.organization
-            and not secret_manager.secret_exists(organization_id, Suffixes.API_KEY)
+            and not user.organization.has_api_key()
```

Comment:
It may be a bit counter intuitive, but `user.organization.has_api_key` is a boolean, not a method. You can drop the `()` at the end.